src/Core/Compute/Metadata.php

```diff
@@ -106,6 +106,25 @@ public function getProjectId()
         return $this->projectId;
     }
 
+    /**
+     * Detect and return the numeric project ID
+     *
+     * Example:
+     * ```
+     * $projectId = $metadata->getNumericProjectId();
+     * ```
+     *
+     * @return string
+     */
+    public function getNumericProjectId()
+    {
+        if (!isset($this->projectId)) {
```

Comment:
This behaves weirdly when you call `getProjectId` first, then `getNumericProjectId`

Reply:
Ah, you're right. I'll fix to memoize to its own variable.